packages/vite/src/vite-node.ts

```diff
@@ -179,3 +183,63 @@ export async function initViteNodeServer (ctx: ViteBuildContext) {
     `export { default } from ${JSON.stringify(pathToFileURL(manifestResolvedPath).href)}`
   )
 }
+
+/**
+ * The following code is ported from vite
+ * Awaits https://github.com/vitejs/vite/pull/12894
+ */
```

Comment:
@pi0 I'd suggest we do nothing about the code below and wait for the Vite PR to be shipped (so we could remove them). Unless you are saying it could be a nice addition to pathe but not related to this PR.

Reply:
Not specific for this PR of course. And only if finally vite can leverage from same util. Otherwise closest impl is best.